TunnelVisionLabs.ReferenceAssemblyAnnotator/CustomAttributeFactory.cs

```diff
@@ -76,5 +76,12 @@ public CustomAttribute AttributeUsage(AttributeTargets validOn, bool? allowMulti
 
             return customAttribute;
         }
+
+        public CustomAttribute ReferenceAssembly()
+        {
+            MethodDefinition constructor = _wellKnownTypes.SystemRuntimeCompilerServicesReferenceAssemblyAttribute.Resolve().Methods.Single(method => method.IsConstructor && !method.IsStatic && method.Parameters.Count == 0);
```

Comment:
❕ This failed in tests:

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at TunnelVisionLabs.ReferenceAssemblyAnnotator.CustomAttributeFactory.ReferenceAssembly()
   at TunnelVisionLabs.ReferenceAssemblyAnnotator.Program.EnsureReferenceAssemblyAttribute(AssemblyDefinition assemblyDefinition, WellKnownTypes wellKnownTypes, CustomAttributeFactory attributeFactory)
   at TunnelVisionLabs.ReferenceAssemblyAnnotator.Program.Main(TaskLoggingHelper log, String referenceAssembly, String annotatedReferenceAssembly, String outputAssembly)
   at TunnelVisionLabs.ReferenceAssemblyAnnotator.AnnotatorBuildTask.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()
```

Reply:
It's first defined in .NET Framework 4.0: https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.referenceassemblyattribute?view=netframework-4.8#applies-to

According to https://github.com/dotnet/roslyn/blob/master/docs/features/refout.md#definition-of-ref-assemblies, `ReferenceAssemblyAttribute` can be defined internally like other attributes.

Reply:
It's first defined in .NET Framework 4.0: https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.referenceassemblyattribute?view=netframework-4.8#applies-to

According to https://github.com/dotnet/roslyn/blob/master/docs/features/refout.md#definition-of-ref-assemblies, `ReferenceAssemblyAttribute` can be defined internally like other attributes.

Reply:
I bet it's not necessary to define the `Description` property or constructor overload or to apply [AttributeUsage].

Reply:
Defining it would only be useful if you compile for net35 or earlier and the output assembly ends up getting used on net40 or later.

Reply:
➡️ I updated the implementation to emit the attribute if it's not already available.